pkg/api/operators/apis/trial/v1alpha2/trial_types.go

```diff
@@ -54,6 +60,23 @@ type TrialStatus struct {
 	Observation Observation `json:"observation,omitempty"`
 }
 
+type ObjectiveSpec struct {
+	Type                ObjectiveType `json:"type,omitempty"`
+	Goal                *float64      `json:"goal,omitempty"`
+	ObjectiveMetricName string        `json:"objectiveMetricName,omitempty"`
+	// This can be empty if we only care about the objective metric.
+	// Note: If we adopt a push instead of pull mechanism, this can be omitted completely.
+	AdditionalMetricNames []string `json:"additionalMetricNames,omitempty"`
+}
+
+type ObjectiveType string
+
+const (
+	ObjectiveTypeUnknown  ObjectiveType = ""
+	ObjectiveTypeMinimize ObjectiveType = "minimize"
+	ObjectiveTypeMaximize ObjectiveType = "maximize"
+)
+
```

Comment:
maybe move above to a common_types.go since they are shared by by trial and experiment

Reply:
Done